### DISCUSSION TITLE
[ZK Stack] - Fee estimation is broken on the Era and Abstract bridge UIs

### DISCUSSION THREAD

**Discussion by cygaar** [zkSync-Community-Hub/zksync-developers#771] [ZK Stack] [answered]
### Team or Project

Abstract

### Module Affected

- [ ] System contracts (era-system-contracts)
- [ ] Smart contracts (era-contracts)
- [ ] zkEVM (era-zk_evm)
- [ ] Server (zksync-era)
- [ ] Prover (era-boojum)
- [ ] Fee withdrawer (era-fee-withdrawer)
- [ ] Block Explorer (clock-explorer)
- [X] Portal - Wallet + Bridge (dapp-portal)
- [ ] Other (please specify below)

### Rust Version

1.75.0

### Operating System

- [ ] Ubuntu
- [X] MacOS
- [ ] Windows
- [ ] Other (Please specify below)

### Issue Description

The portal UI does not let me bridge ETH on either Abstract or zkSync due to insufficient funds errors. However, these errors are wrong because I have 0.4 ETH in my wallet. The interesting part of this is I was able to bridge an ERC20 token without issue.

### Expected Behavior

I should be able to bridge funds to Era or Abstract

### Current Behavior

I'm unable to bridge despite having sufficient funds to cover gas fees.

### Additional Details

![Screenshot 2024-11-11 at 12 04 00 PM](https://github.com/user-attachments/assets/03c5f5f3-717b-4dce-a80a-3d1eaa026c1b)
![Screenshot 2024-11-11 at 12 06 48 PM](https://github.com/user-attachments/assets/bb1894f7-6baf-464e-be5b-b3cf5e5a1635)

### Prior Research

- [x] Yes, I searched for similar issues and did not find any matching ones.

**Comment by alhenaai** [comment]
It appears you are encountering a fee estimation error on the ZKsync Portal while trying to bridge ETH. This issue might be related to the contract function used for fee estimation, which can sometimes revert due to various reasons, such as changes in gas prices or issues within the contract itself.

Here are a few steps you can try to resolve this issue:
1. **Refresh the Portal**: Sometimes, simply refreshing the UI can resolve transient issues.
2. **Clear Browser Cache**: Clearing your browser cache can help if the issue is related to a stale state in the browser.
3. **Check Gas Prices**: Ensure that the gas prices have not spiked unusually. High network congestion can affect transactions and fee estimations.
4. **Try a Different Amount**: If possible, try bridging a different amount of ETH to see if the issue persists.

If the problem continues, it would be beneficial to report this issue directly on the ZKsync developers discussions on Github for further assistance and to alert the development team about this recurring issue. You can post your issue [here](https://github.com/zkSync-Community-Hub/zksync-developers/discussions) for more in-depth support.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by popzxc** [comment]
Hey @cygaar! I'm unable to reproduce it, but from what I see the portal fails to estimate the fee, and this error only seems to be emitted when ethers emits a corresponding [message](https://github.com/matter-labs/dapp-portal/blob/16a5d19b7bbc06974657ae1c603af58c204e059c/composables/zksync/deposit/useFee.ts#L108C39-L108C81), so the origin of the error is in ethers estimation logic. Maybe it picks a gas price that is too high, it is hard to say for sure.

A few follow-up questions:
- Does this error reproduce with other wallets? With other browsers?
- Have you tried disconnecting your wallet, removing the cache, and trying to connect it again?
- Is this problem persistent or sporadic?

**Reply by cygaar** [reply]
I’m using MetaMask - I was able to reproduce on different accounts on MetaMask. I know the zyfi folks were having issues as well.

I’ve tried disconnecting wallet with no luck. The problem is persistent.

**Reply by sachanayush47** [reply]
> I’m using MetaMask - I was able to reproduce on different accounts on MetaMask. I know the zyfi folks were having issues as well.
> 
> I’ve tried disconnecting wallet with no luck. The problem is persistent.

What are the `nominator` and `denominator` values ?

**Reply by popzxc** [reply]
Have you tried resetting the cache in the browser? I'm trying to understand where the issue is, and the only hypothesis I have right now is that maybe fee estimation error was cached and instead of re-estimating it a value from cache is being used (it shouldn't be, but that's a hypothesis).

Or maybe, if you try from a different device with the same MetaMask account, would the issue still reproduce?

**Reply by cygaar** [reply]
I can try resetting the cache

**Reply by cygaar** [reply]
Cleared my cache, still no good. Here are some of the errors I see in the console:
![Screenshot 2024-11-12 at 9 46 41 AM](https://github.com/user-attachments/assets/6bb01b48-1d03-4cd8-b73f-aaac14629aea)

**Reply by popzxc** [reply]
That's weird. We will investigate it further, but I've spent some time trying to find the cause already and failed, so it might take some time.

**Reply by cygaar** [reply]
Are you on slack? Happy to debug in realtime or with faster cadence with you.

**Reply by popzxc** [reply, marked as answer]
The issue has been fixed in https://github.com/matter-labs/dapp-portal/pull/197, thanks for reporting!
